lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -251,4 +232,12 @@ sub finalize_results {
     }
 }
 
+sub _read_text_data ($self, $dir, $filename) {
+    my $text_data = do {
+        try { path($dir, $filename)->slurp }
+        catch ($e) { "Unable to read $filename." }
+    };
+    return decode('UTF-8', $text_data) // "Unable to decode $filename.";
```

Comment:
The question is, should `finalize_results` actually modify the content of the `details-*.json` like this, or should the fallback "Unable to ..." only be presented on the fly when viewing job details?

Reply:
I changed it and left `finalize_results` as it was